sw/device/silicon_creator/rom_ext/rom_ext.c

```diff
@@ -197,8 +202,48 @@ static rom_error_t boot_svc_next_boot_bl0_slot_handler(
   return kErrorRomExtBootFailed;
 }
 
+OT_WARN_UNUSED_RESULT
+static rom_error_t boot_svc_primary_boot_bl0_slot_handler(
+    boot_svc_msg_t *boot_svc_msg, boot_data_t *boot_data) {
+  uint32_t active_slot = boot_data->primary_bl0_slot;
+  uint32_t requested_slot = boot_svc_msg->primary_bl0_slot_req.primary_bl0_slot;
+
+  // In cases where the primary is already set to the requested slot, this
+  // function is a no-op.
+  if (active_slot != requested_slot) {
```

Comment:
nit: wdyt think about hardening this check as well with `if (launder32() { HARDENED_CHECK_NE(...);` and `else { HARDENED_CHECK_EQ(...)`